servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -533,6 +550,16 @@ void markClosed() {
             }
         }
 
+        private ConnState closeConnState() {
+            for (;;) {
+                final ConnState oldState = connState;
+                if (oldState.state == State.CLOSED || connStateUpdater.compareAndSet(this, oldState,
+                        new ConnState(oldState.connections, State.CLOSED))) {
```

Comment:
Consider adding a comment to describe why we need to keep `oldState.connections` around even if `markClosed()` tries to close them. Would be helpful for someone new.